src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -526,4 +526,16 @@ default void setNotifier(Vector3i pos, @Nullable UUID uuid) {
      */
     Set<AABB> getIntersectingCollisionBoxes(Entity owner, AABB box);
 
+    /**
+     * Creates a new archetype volume from the specified section of this extent.
+     * The archetype's volume will be shifted such that the position given in
+     * the origin will be the origin of the volume.
+     *
+     * @param min The minimum point of the area to copy
+     * @param max The maximum point of the area to copy
+     * @param origin The eventual origin on the new archetype volume
```

Comment:
It feels like a bad idea to have the min and max of the archetype volume be mutable and if you change the offset you are also changing the min and max. And for ease of use we want the offset to always be applied to them such that iterating over the volume is always relative to the offset.